cartographer_ros/cartographer_ros/offline_node.cc

```diff
@@ -370,15 +371,19 @@ void RunOfflineNode(const MapBuilderFactory& map_builder_factory) {
   LOG(INFO) << "Peak memory usage: " << usage.ru_maxrss << " KiB";
 #endif
 
-  if (::ros::ok() && bag_filenames.size() > 0) {
-    const std::string output_filename = bag_filenames.front();
-    const std::string suffix = ".pbstream";
-    const std::string state_output_filename = output_filename + suffix;
+  // Serialize unless we have neither a bagfile nor an explicit state filename.
```

Comment:
Same as above, please return the check for ros::ok(). We do not want to serialize if we received a sigabrt.

Reply:
Hmm... but this doesn't happen with `CHECK(ros::ok())`?

Reply:
Isn‘t quitting with a failed CHECK due to ctrl-C a bit ugly than returning cleanly?

Reply:
You have a point that it might be helpful to print an info message saying we are skipping serialization due to node shutting down. But a failed CHECK would still feel to ugly to me.

Reply:
For a batch processing tool like the offline node I consider any case where we run into "ROS not ok" before all the work is done as an error, so I see it as a bug if the offline node **doesn't** return with a non-zero exit code. For example: if the offline node is used in a pipeline and exits silently with "0" although it received a SIGINT in the middle of processing. 

An alternative without checks would be to log an error message and let the executable exit with `EXIT_FAILURE`.

Reply:
Hmm, I see your point, you may be right about the return code. I have been mostly using the offline node as a regular ROS node - I spin it up, look at how it's doing, and maybe quit it if I am not happy. I reacted because I thought that barfing up a failed CHECK error message when you quit it using ctrl-C seemed odd to me - user's desire to quit is not an error per se.

Anyway, I just wanted to point that out, don't really know what really is the proper thing to do here. Maybe the maintainers will know :)

Reply:
> barfing up a failed CHECK error message

I agree that stacktrace and core dump is a bit overdramatic ;) I changed it back for now, it was just something I noticed while digging through the code and not super urgent.

In general the repeated checks for `ros::ok` are a bit odd. If for example final optimization takes very long it still blocks if you ctrl-c until the next `if (!ros::ok) { return; }` is hit.